### PR TITLE
feat: added wallet sync edge case handling

### PR DESCRIPTION
**Pull request by masterchief164**

### CODE REVIEW

File: packages/cysync-core/src/components/DialogManager.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ const getDialogFromState = (state: IDialogState) => {
 
       const dialogState = state[dialogName];
       const DialogComponent = dialogs[dialogName];
-
+      console.log(dialogState.data);
```
Comment (irshadCypherock) [inline packages/cysync-core/src/components/DialogManager.tsx]: Remove console logs

File: packages/cysync-core/src/bgTask/walletSyncTask/index.tsx [hunk 2]
```diff
@@ -1,21 +1,30 @@
...
+  };
+
   useEffect(() => {
     onConnectionChange();
   }, [connection]);
 
+  useEffect(() => {
+    if (deletedWallets.length > 0) onDeleteChange();
+  }, [deletedWallets]);
+
```
Comment (irshadCypherock) [inline packages/cysync-core/src/bgTask/walletSyncTask/index.tsx]: Should we close the dialog if the `deletedWallets`  are 0?

File: packages/ui/src/components/molecules/CheckboxList.tsx [hunk 3]
```diff
@@ -2,7 +2,11 @@ import React, { Dispatch, FC, SetStateAction } from 'react';
 
 import { CheckBox, Flex } from '../atoms';
 
-export type CheckboxListItems = { label: string; checked: boolean }[];
+export type CheckboxListItems = {
+  label: string;
+  checked: boolean;
+  __id: string | undefined;
```
Comment (irshadCypherock) [inline packages/ui/src/components/molecules/CheckboxList.tsx]: Can we just change it to `id`?
```suggestion
 id: string | undefined;
```

File: packages/cysync-core/src/dialogs/WalletSyncError.tsx [hunk 4]
```diff
@@ -1,46 +1,56 @@
...
-        label:
-          lang.strings.walletSync.freshOneCreated.checkboxList.cypherockRed,
-        checked: true,
-      },
-      {
-        label: lang.strings.walletSync.freshOneCreated.checkboxList.official,
-        checked: true,
-      },
-      {
-        label: lang.strings.walletSync.freshOneCreated.checkboxList.personal,
```
Comment (irshadCypherock) [inline packages/cysync-core/src/dialogs/WalletSyncError.tsx:35]: Remove these texts from `i18n/en` as well

File: packages/cysync-core/src/dialogs/WalletSyncError.tsx [hunk 5]
```diff
@@ -1,46 +1,56 @@
...
 
+  const deleteSelectedWallets = () => {
+    const selectedWallets = [] as IWallet[];
+    selectedCheckboxListItems.forEach(
+      item =>
+        item.checked &&
+        selectedWallets.push(
+          deletedWallets.find(wallet => wallet.__id === item.__id)!,
+        ),
+    );
```
Comment (irshadCypherock) [inline packages/cysync-core/src/dialogs/WalletSyncError.tsx]:
```suggestion
    const selectedWallets = selectedCheckboxListItems.filter(
      i => i.checked).map(i => ({ __id: i.id }))
```

File: packages/cysync-core/src/dialogs/WalletSyncError.tsx [hunk 6]
```diff
@@ -1,46 +1,56 @@
...
+
+    dispatch(deleteWallets(selectedWallets));
+    if (deleteWalletStatus === 'succeeded')
+      dispatch(closeDialog('walletSyncError'));
```
Comment (irshadCypherock) [inline packages/cysync-core/src/dialogs/WalletSyncError.tsx]: This is an asynchronous task, so we'll have to listen for changes in the status. We should not depend on it to change immediately after `deleteWallets` dispatch

**Review by irshadCypherock — CHANGES_REQUESTED** [review]
- Add a changeset
- Change the PR title to `feat: added wallet sync edge case handling`

File: packages/cysync-core/src/bgTask/walletSyncTask/index.tsx (hunk 2 above)
Comment (masterchief164) [reply packages/cysync-core/src/bgTask/walletSyncTask/index.tsx]: We haven't opened the dialog yet.

File: packages/cysync-core/src/dialogs/WalletSyncError.tsx (hunk 5 above)
Comment (masterchief164) [reply packages/cysync-core/src/dialogs/WalletSyncError.tsx]: We need the IWallets in the delete wallets function and thus the above refactor won't work.

File: packages/cysync-core/src/components/DialogManager.tsx [hunk 7]
```diff
@@ -17,7 +17,6 @@ const getDialogFromState = (state: IDialogState) => {
 
       const dialogState = state[dialogName];
       const DialogComponent = dialogs[dialogName];
-
```
Comment (irshadCypherock) [inline packages/cysync-core/src/components/DialogManager.tsx]: Any specifics for this change?

File: packages/cysync-core/src/dialogs/WalletSyncError.tsx [hunk 8]
```diff
@@ -1,46 +1,60 @@
...
 
+  const deleteSelectedWallets = () => {
+    const selectedWallets = [] as IWallet[];
+    selectedCheckboxListItems.forEach(
+      item =>
+        item.checked &&
+        selectedWallets.push(
+          deletedWallets.find(wallet => wallet.__id === item.id)!,
+        ),
+    );
```
Comment (irshadCypherock) [inline packages/cysync-core/src/dialogs/WalletSyncError.tsx]: Refactor it to something like this.

```suggestion
    const selectedWalletIds = selectedCheckboxListItems.filter(i => i.checked).map(i => i.id);
    const selectedWallets = deletedWallets.filter(d => selectedWalletIds.includes(d.__id));
```

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

**Review by irshadCypherock — APPROVED** [review]

**Pull request merged**